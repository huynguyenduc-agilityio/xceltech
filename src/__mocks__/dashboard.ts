export const mockPaySlipData = [
  {
    type: 'Basic Wage',
    amount: '150,000',
    deductions: '-30,000',
    total: '120,000',
  },
  { type: 'Tax', amount: '15,000', deductions: '-3,000', total: '12,000' },
  { type: 'Pension', amount: '15,000', deductions: '-3,000', total: '12,000' },
  {
    type: 'Total Earnings',
    amount: '150,000',
    deductions: '-36,000',
    total: '114,000',
  },
];
