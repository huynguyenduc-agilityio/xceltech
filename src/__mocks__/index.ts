export * from './number';
export * from './dashboard';
