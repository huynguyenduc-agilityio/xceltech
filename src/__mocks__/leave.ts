export const mockDataSource = [
  {
    id: '1',
    employeeName: 'John Caster',
    durations: 5,
    startDate: '2025-03-31',
    endDate: '2025-04-30',
    type: 'Sick',
    reason: 'Personal',
  },
  {
    id: '2',
    employeeName: 'aman bey',
    durations: 7,
    startDate: '2025-03-31',
    endDate: '2025-03-31',
    type: 'Exam',
    reason: 'Examination',
  },
  {
    id: '3',
    employeeName: 'feven Tesfaye',
    durations: 120,
    startDate: '2025-03-31',
    endDate: '2025-03-31',
    type: 'Maternity',
    reason: 'Child Care',
  },
];

export const mockDataRecallNotifications = [
  {
    id: 53,
    message:
      'Your Annual leave has been recalled by nhan.tran170495+01@gmail.com. Please return before 2025-04-30.',
    isRead: false,
    createdAt: '2025-04-14T09:18:59.339734Z',
    recallId: '70a67030-2b9f-44cf-8eea-ec61da196b53',
    user: '3c5a91cf-58a4-4dbf-ba47-af1239130b26',
  },
];

export const mockDataRecall = [
  {
    id: '1',
    employeeName: 'John Caster',
    durations: 5,
    startDate: '2025-03-31',
    endDate: '2025-04-30',
    type: 'Sick',
    reason: 'Personal',
  },
  {
    id: '6fbf1cfa-9c29-4c8f-93d6-9f105c644be1',
    startDate: '2025-04-16',
    endDate: '2025-04-30',
    resumptionDate: '2025-04-18',
    employeeName: 'loc vo thanh',
    type: 'Annual',
    employee: '3c5a91cf-58a4-4dbf-ba47-af1239130b26',
    documentPath: null,
    reason: 'Sick',
    durations: 1,
    reliefOfficerFirstName: 'Nhan',
    reliefOfficerLastName: 'Tran',
    status: 'Approved',
    recallStatus: 'Approved',
    recallReason: '',
    recallDate: '2025-04-25',
    isRecalled: true,
    daysRemaining: null,
    createdAt: '2025-04-16T02:22:12.781297Z',
    updatedAt: '2025-04-16T02:49:58.756543Z',
  },
  {
    id: '7bce4e74-1236-42b6-94f5-50206fa5be99',
    startDate: '2025-04-16',
    endDate: '2025-04-30',
    resumptionDate: '2025-05-01',
    employeeName: 'loc vo thanh',
    type: 'Sick',
    employee: '3c5a91cf-58a4-4dbf-ba47-af1239130b26',
    documentPath: null,
    reason: 'Sick',
    durations: 14,
    reliefOfficerFirstName: 'Nhan',
    reliefOfficerLastName: 'Tran',
    status: 'Approved',
    recallStatus: 'Rejected',
    recallReason: "Don't care",
    recallDate: '2025-04-18',
    isRecalled: true,
    daysRemaining: null,
    createdAt: '2025-04-16T02:50:35.646634Z',
    updatedAt: '2025-04-16T02:52:20.564141Z',
  },
];

export const mockDataRecallDetail = {
  id: '6fbf1cfa-9c29-4c8f-93d6-9f105c644be1',
  startDate: '2025-04-16',
  endDate: '2025-04-30',
  resumptionDate: '2025-04-18',
  employeeName: 'loc vo thanh',
  type: 'Annual',
  employee: '3c5a91cf-58a4-4dbf-ba47-af1239130b26',
  documentPath: null,
  reason: 'Sick',
  durations: 1,
  reliefOfficerFirstName: 'string',
  reliefOfficerLastName: 'string',
  status: 'Approved',
  recallStatus: 'Approved',
  recallReason: '',
  recallDate: '',
  isRecalled: false,
  daysRemaining: null,
  createdAt: '2025-04-16T02:22:12.781297Z',
  updatedAt: '2025-04-16T02:49:58.756543Z',
};
