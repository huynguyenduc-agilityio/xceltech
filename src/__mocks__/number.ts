export const minMetricPrefixLimit = 1000;

export const decimalPartMocker = {
  secondDigit: 2,
  thirdDigit: 3,
};

export const numberMocker = {
  hundred: 365,
  thousand: 365259,
  million: 365259000,
  negative: -100,
};
