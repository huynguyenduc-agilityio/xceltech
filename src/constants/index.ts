export * from './navigation';
export * from './unit';
export * from './regex';
export * from './pagination';
export * from './toast';
export * from './form';
export * from './router';
export * from './api';
export * from './config';
export * from './localStorageKey';
export * from './queries';
export * from './field';
export * from './file';
export * from './images';
export { default as MESSAGES } from './message';
