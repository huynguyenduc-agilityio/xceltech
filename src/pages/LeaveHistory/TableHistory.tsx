import { useEffect, useState } from 'react';

// Constants
import {
  DEFAULT_CURRENT_PAGE,
  DEFAULT_PAGE_SIZE,
  DEFAULT_TOTAL,
  MESSAGES,
} from '@/constants';

// Types
import {
  FilterCriteria,
  LeaveHistoryResponse,
  LeaveType,
  StatusLeave,
  TDataSource,
  ToastStatus,
} from '@/types';

// Icons
import { CircleDownIcon } from '@/icons';

// Components
import {
  ActionsDropdown,
  Button,
  DropdownExport,
  FilterMenu,
  Pagination,
  Status,
  Table,
} from '@/components';

// Hooks
import {
  toast,
  useGetLeaves,
  usePagination,
  useUpdateStatusLeaveRequest,
} from '@/hooks';

const TableHistory = () => {
  const [filters, setFilters] = useState<FilterCriteria>();
  const [leaveData, setLeaveData] = useState<LeaveHistoryResponse>();

  const {
    currentPage,
    pageSize,
    handleChangeLimit,
    handleChangePage,
    isDisableNext,
    isDisablePrev,
  } = usePagination(leaveData?.metaData.totalCount);

  const { leaves, isLeavesLoading } = useGetLeaves({
    page: currentPage,
    limit: pageSize,
    filters: { ...filters },
  });

  const { handleUpdateStatus } = useUpdateStatusLeaveRequest({
    page: currentPage,
    limit: pageSize,
    filters: { ...filters },
  });

  const handleApplyFilters = (filters: FilterCriteria) => {
    setFilters(filters);
    handleChangePage(DEFAULT_CURRENT_PAGE);
  };

  const handleActionsLeave = async (leaveId: string, status: StatusLeave) => {
    try {
      await handleUpdateStatus({ leaveId, status });

      toast({
        status: ToastStatus.Success,
        title: MESSAGES.COMMON.LEAVE_REQUEST_SUCCESS,
      });
    } catch (error) {
      toast({
        status: ToastStatus.Error,
        title: MESSAGES.COMMON.LEAVE_REQUEST_FAILED,
      });
    }
  };

  useEffect(() => {
    if (leaves) {
      setLeaveData(leaves);
    }
  }, [leaves]);

  const renderActionsDropdown = ({ id, status }: TDataSource) => {
    const isDisabled = status !== StatusLeave.Pending;

    return (
      <ActionsDropdown
        items={[
          {
            label: 'Approved',
            onClick: () => handleActionsLeave(id, StatusLeave.Approved),
          },
          {
            label: 'Rejected',
            onClick: () => handleActionsLeave(id, StatusLeave.Rejected),
          },
        ]}
      >
        <Button disabled={isDisabled} className="w-[195px] h-14 shadow-md">
          Action
          <CircleDownIcon className="ml-4" />
        </Button>
      </ActionsDropdown>
    );
  };

  const renderColumns = [
    {
      title: 'Name(s)',
      key: 'employeeName',
    },
    {
      title: 'Duration(s)',
      key: 'durations',
    },
    {
      title: 'Start Date',
      key: 'startDate',
    },
    {
      title: 'End Date',
      key: 'endDate',
    },
    {
      title: 'Type',
      key: 'type',
    },
    {
      title: 'Status',
      key: 'status',
      renderBody: ({ status }: TDataSource) => (
        <div className="flex items-center justify-center">
          <Status type={status as StatusLeave} />
        </div>
      ),
    },
    {
      title: 'Actions',
      key: 'action',
      renderBody: renderActionsDropdown,
    },
  ];
  return (
    <div>
      <div className="flex justify-between px-5 pt-0 py-6">
        <h2 className="text-xl font-bold">Leave History</h2>
        <div className="flex items-center gap-14">
          <FilterMenu
            options={{ type: Object.values(LeaveType) }}
            onApply={handleApplyFilters}
          />
          <DropdownExport />
        </div>
      </div>

      <Table
        columns={renderColumns}
        dataSource={leaveData?.results}
        isLoading={isLeavesLoading}
      />
      <Pagination
        currentPage={currentPage || DEFAULT_CURRENT_PAGE}
        limit={pageSize || DEFAULT_PAGE_SIZE}
        totalRecords={leaveData?.metaData.totalCount || DEFAULT_TOTAL}
        onChangeLimit={handleChangeLimit}
        onPageChange={handleChangePage}
        isDisabledNext={isDisableNext}
        isDisabledPrev={isDisablePrev}
      />
    </div>
  );
};

export default TableHistory;
