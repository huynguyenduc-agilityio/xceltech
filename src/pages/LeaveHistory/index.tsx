import TableHistory from './TableHistory';

const LeaveHistory = () => (
  <div className="bg-white p-12 mt-8 rounded-3xl">
    <TableHistory />
  </div>
);

export default LeaveHistory;
