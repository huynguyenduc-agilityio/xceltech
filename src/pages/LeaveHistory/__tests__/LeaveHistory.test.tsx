import { ReactNode } from 'react';
import userEvent from '@testing-library/user-event';
import { render, RenderResult, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Pages
import LeaveHistory from '..';

// Mocks
import { mockDataSource } from '@/__mocks__';

// Constants
import { MESSAGES } from '@/constants';

let renderResult: RenderResult;
const queryClient = new QueryClient();

const mockUseGetLeaves = jest.fn();
const mockUsePagination = jest.fn();
const mockUseUpdateStatusLeaveRequest = jest.fn();

jest.mock('@/hooks', () => ({
  ...jest.requireActual('@/hooks'),
  useGetLeaves: () => mockUseGetLeaves(),
  usePagination: () => mockUsePagination(),
  useUpdateStatusLeaveRequest: () => mockUseUpdateStatusLeaveRequest(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useSearchParams: () => [new URLSearchParams()],
}));

const Wrapper = ({ children }: { children: ReactNode }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

describe('LeaveHistory', () => {
  beforeEach(() => {
    mockUseGetLeaves.mockReturnValue({
      leaves: {
        results: mockDataSource,
        metaData: { totalCount: mockDataSource.length },
      },
      isLeavesLoading: false,
    });

    mockUsePagination.mockReturnValue({
      currentPage: 1,
      pageSize: 10,
      handleChangeLimit: jest.fn(),
      handleChangePage: jest.fn(),
      isDisableNext: false,
      isDisablePrev: false,
    });

    mockUseUpdateStatusLeaveRequest.mockReturnValue({
      isLoading: false,
      handleUpdateStatus: jest.fn(),
    });

    renderResult = render(
      <Wrapper>
        <LeaveHistory />
      </Wrapper>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render successfully', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('handle click button actions', async () => {
    const { getAllByRole, getAllByText, container } = renderResult;

    const actionButton = getAllByRole('button', { name: 'Action' })[0];

    await userEvent.click(actionButton);

    await waitFor(async () => {
      const approvedButton = getAllByText('Approved')[0];
      await userEvent.click(approvedButton);
    });

    await userEvent.click(actionButton);

    await waitFor(async () => {
      const declinedButton = getAllByText('Rejected')[0];

      await userEvent.click(declinedButton);
    });

    expect(container).toBeInTheDocument();
  });

  it('handle click filter dropdown', async () => {
    const { getByTitle, getByRole, container } = renderResult;

    const filterButton = getByTitle('Filter Dropdown');

    await userEvent.click(filterButton);

    await waitFor(async () => {
      const applyButton = getByRole('button', { name: 'OK' });

      await userEvent.click(applyButton);
    });

    expect(container).toBeInTheDocument();
  });

  it('should render empty state when there is no data', () => {
    mockUseGetLeaves.mockReturnValue({
      leaves: {
        results: [],
        metaData: { totalCount: 0 },
      },
      isLeavesLoading: false,
    });

    mockUsePagination.mockReturnValue({
      currentPage: null,
      pageSize: null,
      handleChangeLimit: jest.fn(),
      handleChangePage: jest.fn(),
      isDisableNext: false,
      isDisablePrev: false,
    });

    const { getByText } = render(
      <Wrapper>
        <LeaveHistory />
      </Wrapper>,
    );

    expect(getByText(MESSAGES.COMMON.EMPTY_DATA)).toBeInTheDocument();
  });
});
