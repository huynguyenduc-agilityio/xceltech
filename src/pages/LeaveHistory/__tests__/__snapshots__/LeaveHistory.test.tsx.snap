// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveHistory should render successfully 1`] = `
<div>
  <div
    class="bg-white p-12 mt-8 rounded-3xl"
  >
    <div>
      <div
        class="flex justify-between px-5 pt-0 py-6"
      >
        <h2
          class="text-xl font-bold"
        >
          Leave History
        </h2>
        <div
          class="flex items-center gap-14"
        >
          <button
            aria-controls="radix-:r0:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-full p-2"
            data-state="closed"
            title="Filter Dropdown"
            type="button"
          >
            <svg
              fill="none"
              height="25"
              viewBox="0 0 31 25"
              width="31"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M29.545 0H1.455C.165 0-.487 1.263.427 2l11.198 9.032v10.062c0 .382.231.74.62.96l4.844 2.733c.955.54 2.286-.007 2.286-.96V11.032l11.198-9.031C31.486 1.264 30.839 0 29.545 0"
                  fill="#1A1A1A"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h31v25H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            aria-expanded="false"
            aria-haspopup="menu"
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed text-white disabled:hover:bg-primary py-6 w-[195px] h-14 bg-green-primary hover:bg-green-primary/90 shadow-md"
            data-state="closed"
            id="radix-:r1:"
            type="button"
          >
            Export
            <svg
              class="ml-4"
              fill="none"
              height="14"
              viewBox="0 0 17 14"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M16.734 7c0 3.746-3.685 6.781-8.234 6.781S.266 10.746.266 7 3.95.219 8.5.219 16.734 3.254 16.734 7m-7.67 3.114 4.5-3.705c.312-.257.312-.672 0-.927l-.565-.464a.92.92 0 0 0-1.126 0L8.5 7.796 5.127 5.018a.92.92 0 0 0-1.126 0l-.564.464c-.313.257-.313.673 0 .927l4.499 3.706a.92.92 0 0 0 1.128 0"
                  fill="#fff"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 0h17v14H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </div>
      </div>
      <div
        class="relative w-full overflow-auto"
      >
        <table
          class="w-full caption-bottom text-sm"
        >
          <thead
            class="text-lg font-bold"
          >
            <tr
              class="transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted bg-blue-light"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Name(s)
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Duration(s)
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Start Date
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                End Date
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Type
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Status
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Actions
              </td>
            </tr>
          </thead>
          <tbody
            class="[&_tr:last-child]:border-0 text-lg"
          >
            <tr
              class="transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted px-5 bg-white"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                John Caster
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                5
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-03-31
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-04-30
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Sick
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <div
                  class="flex items-center justify-center"
                >
                  <div
                    class="flex items-center w-28 justify-center gap-1 px-2.5 py-1 rounded-full border text-white bg-gray-500"
                  >
                    <span
                      class="text-base font-bold capitalize"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white disabled:hover:bg-primary py-6 w-[195px] h-14 shadow-md"
                  data-state="closed"
                  disabled=""
                  id="radix-:r8:"
                  type="button"
                >
                  Action
                  <svg
                    class="ml-4"
                    fill="none"
                    height="14"
                    viewBox="0 0 17 14"
                    width="17"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#a)"
                    >
                      <path
                        d="M16.734 7c0 3.746-3.685 6.781-8.234 6.781S.266 10.746.266 7 3.95.219 8.5.219 16.734 3.254 16.734 7m-7.67 3.114 4.5-3.705c.312-.257.312-.672 0-.927l-.565-.464a.92.92 0 0 0-1.126 0L8.5 7.796 5.127 5.018a.92.92 0 0 0-1.126 0l-.564.464c-.313.257-.313.673 0 .927l4.499 3.706a.92.92 0 0 0 1.128 0"
                        fill="#fff"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="a"
                      >
                        <path
                          d="M0 0h17v14H0z"
                          fill="#fff"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </button>
              </td>
            </tr>
            <tr
              class="transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted px-5 bg-blue-light"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                aman bey
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                7
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-03-31
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-03-31
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Exam
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <div
                  class="flex items-center justify-center"
                >
                  <div
                    class="flex items-center w-28 justify-center gap-1 px-2.5 py-1 rounded-full border text-white bg-gray-500"
                  >
                    <span
                      class="text-base font-bold capitalize"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white disabled:hover:bg-primary py-6 w-[195px] h-14 shadow-md"
                  data-state="closed"
                  disabled=""
                  id="radix-:ra:"
                  type="button"
                >
                  Action
                  <svg
                    class="ml-4"
                    fill="none"
                    height="14"
                    viewBox="0 0 17 14"
                    width="17"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#a)"
                    >
                      <path
                        d="M16.734 7c0 3.746-3.685 6.781-8.234 6.781S.266 10.746.266 7 3.95.219 8.5.219 16.734 3.254 16.734 7m-7.67 3.114 4.5-3.705c.312-.257.312-.672 0-.927l-.565-.464a.92.92 0 0 0-1.126 0L8.5 7.796 5.127 5.018a.92.92 0 0 0-1.126 0l-.564.464c-.313.257-.313.673 0 .927l4.499 3.706a.92.92 0 0 0 1.128 0"
                        fill="#fff"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="a"
                      >
                        <path
                          d="M0 0h17v14H0z"
                          fill="#fff"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </button>
              </td>
            </tr>
            <tr
              class="transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted px-5 bg-white"
            >
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                feven Tesfaye
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                120
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-03-31
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                2025-03-31
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                Maternity
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <div
                  class="flex items-center justify-center"
                >
                  <div
                    class="flex items-center w-28 justify-center gap-1 px-2.5 py-1 rounded-full border text-white bg-gray-500"
                  >
                    <span
                      class="text-base font-bold capitalize"
                    >
                      Pending
                    </span>
                  </div>
                </div>
              </td>
              <td
                class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] py-6 text-center"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-primary text-white disabled:hover:bg-primary py-6 w-[195px] h-14 shadow-md"
                  data-state="closed"
                  disabled=""
                  id="radix-:rc:"
                  type="button"
                >
                  Action
                  <svg
                    class="ml-4"
                    fill="none"
                    height="14"
                    viewBox="0 0 17 14"
                    width="17"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#a)"
                    >
                      <path
                        d="M16.734 7c0 3.746-3.685 6.781-8.234 6.781S.266 10.746.266 7 3.95.219 8.5.219 16.734 3.254 16.734 7m-7.67 3.114 4.5-3.705c.312-.257.312-.672 0-.927l-.565-.464a.92.92 0 0 0-1.126 0L8.5 7.796 5.127 5.018a.92.92 0 0 0-1.126 0l-.564.464c-.313.257-.313.673 0 .927l4.499 3.706a.92.92 0 0 0 1.128 0"
                        fill="#fff"
                      />
                    </g>
                    <defs>
                      <clippath
                        id="a"
                      >
                        <path
                          d="M0 0h17v14H0z"
                          fill="#fff"
                        />
                      </clippath>
                    </defs>
                  </svg>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="w-full flex justify-between mt-6"
      >
        <div
          class="flex items-center"
        >
          <p
            class="text-md"
          >
            1 - 3 of 3 Leaves
          </p>
        </div>
        <div
          class="flex items-center gap-10"
        >
          <div
            class="flex items-center gap-3 text-md"
          >
            <p>
              Rows per page:
            </p>
            <button
              aria-autocomplete="none"
              aria-controls="radix-:r3:"
              aria-expanded="false"
              class="font-productsans flex h-full items-center justify-between px-3 py-2 text-lg ring-offset-background data-[placeholder]:text-gray-soft focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-white border border-gray-dark rounded-md w-fit"
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                10
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground bg-primary rounded-md w-7 h-7 p-0"
              title="prev-button"
            >
              <svg
                class="lucide lucide-move-left text-white w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 8L2 12L6 16"
                />
                <path
                  d="M2 12H22"
                />
              </svg>
            </button>
            <button
              class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground bg-primary rounded-md w-7 h-7 p-0"
              title="next-button"
            >
              <svg
                class="lucide lucide-move-right text-white w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 8L22 12L18 16"
                />
                <path
                  d="M2 12H22"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
